go/pools/smartconnpool/waitlist_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package smartconnpool
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitlistExpireWithMultipleWaiters(t *testing.T) {
```

Comment:
I'm assuming that this test fails on main. Just double checking 🙂

Reply:
Yup, looks like this:
```
=== RUN   TestWaitlistExpireWithMultipleWaiters
    waitlist_test.go:61:
        	Error Trace:	/Users/brendan/src/github.com/Shopify/vitess/go/pools/smartconnpool/waitlist_test.go:61
        	Error:      	Timed out waiting for all waiters to expire
        	Test:       	TestWaitlistExpireWithMultipleWaiters
        	Messages:   	Wanted 2, got 1
```